### DISCUSSION TITLE
Show records in new page

### DISCUSSION THREAD

**Discussion by Hajelarohit** [Budibase/budibase#5625] [General]
I have a table in bb database with multiple fields. I am also able to display it with select few fields on a page. 

What i need is, when i click on a record, the record should open in a new page. I also want to show only few columns from the original table from the database. 

How can i achieve it?

**Comment by oitkenmen** [comment]
- Select the table.
- Select "Link Table Rows" in the right menu.
- Write the link to a new page in the "Link Screen" field.
- In the "ID column for linking" select the column you want to be appended to URL, for example, "username". So that when you click in the row - the new page opens with .../(someusername)

- Build the UI on the new page with containers, and headlines, and various paragraphs that show the columns you want.
- Wrap everything in the Data Provider container. In the Data Provider menu choose the corresponding table. Define filters where (for example) "username" equals {{ URL.id }}. Limit to 1. Under the Data Provider put Repeater.

- Or you can just look into the ready-made templates. They have good examples.

**Reply by melohagan** [reply]
These doc pages should also help: 
- https://docs.budibase.com/docs/url-parameters#link-navigation
- https://docs.budibase.com/docs/table#configure-columns

**Comment by Hajelarohit** [comment]
Thanks Both. It was helpful.

The second challenge i am facing is. My app has multiple users within multiple teams that access the same data. However I don't want to show the entire data to everyone. I want the user to only see that data which is assigned to him because they have to work on it.

e.g. See the table below

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



Team 1 | Team 2 | Team 3 | Engagement Number | Engagement Name
-- | -- | -- | -- | --
User 1 | User 5 | User 8 | 123 | ABC
User 1 | User 6 | User 8 | 234 | BCD
User 2 | User 6 | User 8 | 345 | CDE
User 2 | User 6 | User 8 | 456 | DEF
User 3 | User 6 | User 8 | 567 | EFG
User 3 | User 7 | User 9 | 678 | FGH
User 3 | User 7 | User 10 | 789 | GHI



</body>

</html>

When User 1 logs in he is only able to view/edit first 2 records. When user 2 logs in, he is able to view/edit next two records. When user 6 logs in, he is able to view/edit only records 2-5 so on and so forth. I can use dynamic filter, but I am looking for a more dynamic solution. Not sure how to achieve this.

**Reply by oitkenmen** [reply]
In the right menu of the table, select "Define Filters". Add a filter where (for example) "username" equals {{ Current User.Username }} .

In the table itself, create a column called (for example) "username" and assign usernames of Users to the columns you want.

Thus, the table will show the row only if the "username" column in this row equals to the User's username.

You can also select "Configure Conditions" if you want your table to change its state depending on some condition or action.
